[dotnet] Update troubleshoot documentation (#16463)
DataDog/documentation: content/en/profiler/profiler_troubleshooting.md

   5. Check the other HTTP codes for possible errors such as 403 for invalid API key.

> 4. For missing CPU or Wall time profiles only, check the Datadog signal handler for stack walk has not been replaced:

Author: Would the following be better?

For missing CPU or Wall time profiles only, check **that** the Datadog signal handler for stack walk that has not been replaced:

> 4. For missing CPU or Wall time profiles only, check the Datadog signal handler for stack walk has not been replaced:

Reviewer: ```suggestion
4. For missing CPU or Wall time profiles only, check that the Datadog signal handler for stack walk has not been replaced:
```

   1. Open the `DD-DotNet-Profiler-Native-<Application Name>-<pid>` log file in the `/var/log/datadog` folder.

   2. Look for these two messages:

>       - `Profiler signal handler was replaced again. As of now, we will stopped restoring it to avoid issues: the profiler is disabled.`

Reviewer: ```suggestion
      - `Profiler signal handler was replaced again. It will not be restored: the profiler is disabled.`
```

You can override the 1 core threshold by setting `DD_PROFILING_MIN_CORES_THRESHOLD` environment variable to a value smaller than 1. For example, a value of `0.5` allows the profiler to run in a container with at least 0.5 cores.

## Hang application on Linux

It might happen that an application becomes unresponsive on Linux and CPU/Wall time samples are no more available. In that case, follow these steps:

1. Open the `DD-DotNet-Profiler-Native-<Application Name>-<pid>` log file in the `/var/log/datadog/dotnet` folder.

2. Look for `StackSamplerLoopManager::WatcherLoopIteration - Deadlock intervention still in progress for thread ...`. If this message is not present, the rest does not apply.

3. If the message is found, it means that the stack walking mechanism could have deadlocked. To help investigate the issue, it is very helpful to dump the call stacks of all threads in the application. This is possible with a debugger such as gdb:

   1. Install gdb.

>    2. Type the following command:

Reviewer: ```suggestion
   2. Run the following command:
```

      ```
      gdb -p <process id> -batch -ex "thread apply all bt full" -ex "detach" -ex "quit"
      ```

>    3. Send the resulting output to the Datadog support.

Reviewer: ```suggestion
   3. Send the resulting output to [Datadog Support][2].
```
